[BUG] test_part_write_round_trip[Float] Failed on -0.0 partition

**Describe the bug**
This is essentially the exact same error as in https://github.com/NVIDIA/spark-rapids/issues/9837 but with a -0.0/0.0 instead of NaN. I don't know if we just want to avoid partition tests on floats/doubles all together or if we need to fix this for -0.0/0.0 too.

```
[2023-12-06T16:06:02.846Z] E                   Caused by: org.apache.hadoop.ipc.RemoteException(org.apache.hadoop.fs.FileAlreadyExistsException): /tmp/pyspark_tests/rl-r7525-d32-u27.raplab.nvidia.com-main-108896-70407427/PARQUET_DATA/CPU/_temporary/0/_temporary/attempt_202312061600358867732713042039013_0525_m_000000_603/a=-0.0/part-00000-fa931816-6b21-46b6-a35a-f1aa01858736.c000.snappy.parquet for client 10.150.166.216 already exists
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.hdfs.server.namenode.FSDirWriteFileOp.startFile(FSDirWriteFileOp.java:388)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.startFileInt(FSNamesystem.java:2576)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.hdfs.server.namenode.FSNamesystem.startFile(FSNamesystem.java:2473)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.hdfs.server.namenode.NameNodeRpcServer.create(NameNodeRpcServer.java:773)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolServerSideTranslatorPB.create(ClientNamenodeProtocolServerSideTranslatorPB.java:490)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.hdfs.protocol.proto.ClientNamenodeProtocolProtos$ClientNamenodeProtocol$2.callBlockingMethod(ClientNamenodeProtocolProtos.java)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.ProtobufRpcEngine$Server$ProtoBufRpcInvoker.call(ProtobufRpcEngine.java:533)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.RPC$Server.call(RPC.java:1070)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:989)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.Server$RpcCall.run(Server.java:917)
[2023-12-06T16:06:02.846Z] E                   	at java.security.AccessController.doPrivileged(Native Method)
[2023-12-06T16:06:02.846Z] E                   	at javax.security.auth.Subject.doAs(Subject.java:422)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1898)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.Server$Handler.run(Server.java:2894)
[2023-12-06T16:06:02.846Z] E                   
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.Client.getRpcResponse(Client.java:1597)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.Client.call(Client.java:1543)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.Client.call(Client.java:1440)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:233)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:118)
[2023-12-06T16:06:02.846Z] E                   	at com.sun.proxy.$Proxy32.create(Unknown Source)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.hdfs.protocolPB.ClientNamenodeProtocolTranslatorPB.create(ClientNamenodeProtocolTranslatorPB.java:375)
[2023-12-06T16:06:02.846Z] E                   	at sun.reflect.GeneratedMethodAccessor48.invoke(Unknown Source)
[2023-12-06T16:06:02.846Z] E                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2023-12-06T16:06:02.846Z] E                   	at java.lang.reflect.Method.invoke(Method.java:498)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:431)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeMethod(RetryInvocationHandler.java:166)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invoke(RetryInvocationHandler.java:158)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.io.retry.RetryInvocationHandler$Call.invokeOnce(RetryInvocationHandler.java:96)
[2023-12-06T16:06:02.846Z] E                   	at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:362)
[2023-12-06T16:06:02.846Z] E                   	at com.sun.proxy.$Proxy33.create(Unknown Source)
```

--- a/integration_tests/src/main/python/orc_write_test.py
+++ b/integration_tests/src/main/python/orc_write_test.py
@@ -103,7 +103,7 @@ def test_write_round_trip_corner(spark_tmp_path, orc_gen, orc_impl):
             conf={'spark.sql.orc.impl': orc_impl, 'spark.rapids.sql.format.orc.write.enabled': True})
 
 orc_part_write_gens = [
-        byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen, boolean_gen,
+        byte_gen, short_gen, int_gen, long_gen, boolean_gen,
         # Some file systems have issues with UTF8 strings so to help the test pass even there
         StringGen('(\\w| ){0,50}'),
         # Once https://github.com/NVIDIA/spark-rapids/issues/139 is fixed replace this with
@@ -118,14 +118,7 @@ orc_part_write_gens = [
 @pytest.mark.parametrize('orc_gen', orc_part_write_gens, ids=idfn)
 @pytest.mark.xfail(condition = is_not_utc(), reason = 'xfail non-UTC time zone tests because of https://github.com/NVIDIA/spark-rapids/issues/9653')
 def test_part_write_round_trip(spark_tmp_path, orc_gen):
-    part_gen = orc_gen
-    # Avoid generating NaNs for partition values.
-    # Spark does not handle partition switching properly since NaN != NaN.
-    if isinstance(part_gen, FloatGen):
-        part_gen = FloatGen(no_nans=True)
-    elif isinstance(part_gen, DoubleGen):
-        part_gen = DoubleGen(no_nans=True)
-    gen_list = [('a', RepeatSeqGen(part_gen, 10)),
+    gen_list = [('a', RepeatSeqGen(orc_gen, 10)),
                 ('b', orc_gen)]
     data_path = spark_tmp_path + '/ORC_DATA'
     assert_gpu_and_cpu_writes_are_equal_collect(

--- a/integration_tests/src/main/python/parquet_write_test.py
+++ b/integration_tests/src/main/python/parquet_write_test.py
@@ -164,7 +164,7 @@ def test_write_ts_millis(spark_tmp_path, ts_type, ts_rebase):
 
 
 parquet_part_write_gens = [
-    byte_gen, short_gen, int_gen, long_gen, float_gen, double_gen,
+    byte_gen, short_gen, int_gen, long_gen,
     # Some file systems have issues with UTF8 strings so to help the test pass even there
     StringGen('(\\w| ){0,50}'),
     boolean_gen, date_gen,
@@ -176,14 +176,7 @@ parquet_part_write_gens = [
 @pytest.mark.parametrize('parquet_gen', parquet_part_write_gens, ids=idfn)
 @pytest.mark.xfail(condition = is_not_utc(), reason = 'xfail non-UTC time zone tests because of https://github.com/NVIDIA/spark-rapids/issues/9653')
 def test_part_write_round_trip(spark_tmp_path, parquet_gen):
-    part_gen = parquet_gen
-    # Avoid generating NaNs for partition values.
-    # Spark does not handle partition switching properly since NaN != NaN.
-    if isinstance(part_gen, FloatGen):
-        part_gen = FloatGen(no_nans=True)
-    elif isinstance(part_gen, DoubleGen):
-        part_gen = DoubleGen(no_nans=True)
-    gen_list = [('a', RepeatSeqGen(part_gen, 10)),
+    gen_list = [('a', RepeatSeqGen(parquet_gen, 10)),
                 ('b', parquet_gen)]
     data_path = spark_tmp_path + '/PARQUET_DATA'
     assert_gpu_and_cpu_writes_are_equal_collect(

